File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.js
function solution(str, skip, index) {
    let strArr = str.split('');
    let strNumArr = strArr.map((item) => item.charCodeAt())
    
    let skipArr = skip.split('');
    let skipNumArr = skipArr.map((item) => item.charCodeAt())
    
    for(let j = 0; j < strNumArr.length; j++){
        for(let i = 0; i < index; i++){
            if(strNumArr[j] === 'z'.charCodeAt()){
                strNumArr[j] = 'a'.charCodeAt();
            } else{
                strNumArr[j]++;
            }
            while(skipNumArr.includes(strNumArr[j])){
                if(strNumArr[j] === 'z'.charCodeAt()){
                    strNumArr[j] = 'a'.charCodeAt();
                } else{
                    strNumArr[j]++;
                }
            }
        }
    }
    let resultArr = strNumArr.map((item) => String.fromCharCode(item));
    return resultArr.join('')
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    ans = ''
    
    for i in s:
        now = ord(i)
        
        n = 0
        while n < index:
            now += 1
            
            if now > ord('z'):
                now = ord('a')
            
            if chr(now) in skip:
                continue
            
            n += 1
        
        ans += chr(now)
    return ans
    
'''    
    "aukks"	"wbqd"	5	"happy"

    a -> [b] c [d] e f g h
    u -> v [w] x y z a
    k -> l m n o p
    k -> l m n o p
    s -> t u v [w] x y
'''